frontend_tests/puppeteer_tests/03-compose.js

```diff
@@ -0,0 +1,168 @@
+    await message_we_reply_to.click();
+    await common.check_form_contents(page,
+                                     '#send_message_form',
+                                     {
+                                         stream_message_recipient_stream: "Verona",
+                                         stream_message_recipient_topic: "Reply test",
+                                         content: "",
+                                     });
+    await close_compose_box(page);
+
+    // Check the same for a private message.
+    const priv_msg_we_reply_to = (await page.$x("//p[text()='And reply to this private message']")).slice(-1)[0];
+    await priv_msg_we_reply_to.click();
+    await page.waitForSelector("#private_message_recipient", {visible: true});
+    await common.pm_recipient.expect(page, "user8@zulip.testserver");
+
+    // The last message(private) in the narrow is selected.
+    // Now we go up and open compose box with r key
+    await page.keyboard.press("Escape");
+    await page.keyboard.press("KeyK");
+    await page.keyboard.press("KeyR");
+    await common.check_form_contents(page, '#send_message_form', {
+        stream_message_recipient_stream: "Verona",
+        stream_message_recipient_topic: "Reply test",
+        content: "",
+    });
+
+    // Test opening and closing of compose box.
+    await page.waitForSelector('#stream-message', {hidden: false});
+    await close_compose_box(page);
+    await page.waitForSelector('#stream-message', {hidden: true});
+
+    await page.keyboard.press('KeyX');
+    await page.waitForSelector('#private-message', {hidden: false});
+    await close_compose_box(page);
+    await page.waitForSelector('#private-message', {hidden: true});
+
+
+    // Test focus after narrowing to PMs with a user and typing 'c'
+    const you_and_cordelia_selector = '*[title="Narrow to your private messages with Cordelia Lear"]';
+    // For some unknown reason page.click() isn't working here.
+    await page.evaluate((selector) => document.querySelector(selector).click(),
+                        you_and_cordelia_selector);
+    const cordelia_user_id = 8;
+    const pm_list_selector = `li[data-user-ids-string="${cordelia_user_id}"].expanded_private_message.active-sub-filter`;
+    await page.waitForSelector(pm_list_selector, {visible: true});
+    await close_compose_box(page);
+    await page.keyboard.press('KeyC');
+    await page.waitForSelector('#compose', {visible: true});
+    await page.waitForFunction(() => {
+        return document.activeElement === $('.compose_table #stream_message_recipient_stream')[0];
+    });
+    await close_compose_box(page);
+
+    // Make sure multiple PM recipients display properly.
+    const recipients = ['cordelia@zulip.com', 'othello@zulip.com'];
+    const multiple_recipients_pm = 'A huddle to check spaces';
+    const jquery_selector_pm = ".messagebox:contains('" + multiple_recipients_pm + "')";
```

Comment:
Maybe just `pm_selector`? `jQuery` in the name doesn't add anything (I don't think emphasizing that the selector only works for jQuery is important here). And use string literals to make it more readable.
```suggestion
    const selector_pm= `.messagebox:contains('${multiple_recipients_pm}')`;
```